components/server/src/config/configuration-service.ts

```diff
@@ -36,16 +36,22 @@ export class ConfigurationService {
         }
         // eagerly fetch for all files that the inferrer usually asks for.
         this.requestedPaths.forEach(path => !(path in cache) && readFile(path));
-        const config: WorkspaceConfig = await new ConfigInferrer().getConfig({
+        const configInferrer = new ConfigInferrer();
+        const config: WorkspaceConfig = await configInferrer.getConfig({
             config: {},
             read: readFile,
             exists: async (path: string) => !!(await readFile(path)),
         });
         if (!config.tasks) {
             return;
         }
-        const configString = `tasks:\n  - ${config.tasks.map(task => Object.entries(task).map(([phase, command]) => `${phase}: ${command}`).join('\n    ')).join('\n  - ')}`;
-        return configString;
+        const configString = configInferrer.toYaml(config);
+        return `# This configuration file was automatically generated by Gitpod.
+# Please adjust to your needs (see https://www.gitpod.io/docs/config-gitpod-file)
+# and commit this file to your remote git repository to share the goodness with others.
```

Comment:
I'm not sure about the phrasing "to share the goodness with others", but I don't have a better proposal right now.

@gtsiolis Would you have any suggestions to improve this comment? (It's added to every auto-generated `.gitpod.yml` when you open a Gitpod workspace for any repository that doesn't already have one.)

Note: Let's not block this PR, but work on a better comment in a follow-up PR.